setup.cfg

```diff
@@ -1,10 +1,219 @@
+    foolscap >= 21.7.0
+
+    # * cryptography 2.6 introduced some ed25519 APIs we rely on.  Note that
+    #   Twisted[conch] also depends on cryptography and Twisted[tls]
+    #   transitively depends on cryptography.  So it's anyone's guess what
+    #   version of cryptography will *really* be installed.
+    cryptography >= 2.6
+
+    # * The SFTP frontend depends on Twisted 11.0.0 to fix the SSH server
+    #   rekeying bug <https://twistedmatrix.com/trac/ticket/4395>
+    # * The SFTP frontend and manhole depend on the conch extra. However, we
+    #   can't explicitly declare that without an undesirable dependency on gmpy,
+    #   as explained in ticket #2740.
+    # * Due to a setuptools bug, we need to declare a dependency on the tls
+    #   extra even though we only depend on it via foolscap.
+    # * Twisted >= 15.1.0 is the first version that provided the [tls] extra.
+    # * Twisted-16.1.0 fixes https://twistedmatrix.com/trac/ticket/8223,
+    #   which otherwise causes test_system to fail (DirtyReactorError, due to
+    #   leftover timers)
+    # * Twisted-16.4.0 introduces `python -m twisted.trial` which is needed
+    #   for coverage testing
+    # * Twisted 16.6.0 drops the undesirable gmpy dependency from the conch
+    #   extra, letting us use that extra instead of trying to duplicate its
+    #   dependencies here.  Twisted[conch] >18.7 introduces a dependency on
+    #   bcrypt.  It is nice to avoid that if the user ends up with an older
+    #   version of Twisted.  That's hard to express except by using the extra.
+    #
+    # * Twisted 18.4.0 adds `client` and `host` attributes to `Request` in the
+    # * initializer, needed by logic in our custom `Request` subclass.
+    #
+    #   In a perfect world, Twisted[conch] would be a dependency of an sftp
+    #   extra.  However, pip fails to resolve the dependencies all
+    #   dependencies when asked for Twisted[tls] *and* Twisted[conch].
+    #   Specifically, Twisted[conch] (as the later requirement) is ignored.
+    #   If there were an Tahoe-LAFS sftp extra that dependended on
+    #   Twisted[conch] and install_requires only included Twisted[tls] then
+    #   `pip install tahoe-lafs[sftp]` would not install requirements
+    #   specified by Twisted[conch].  Since this would be the *whole point* of
+    #   an sftp extra in Tahoe-LAFS, there is no point in having one.
+    # * Twisted 19.10 introduces Site.getContentFile which we use to get
+    #   temporary upload files placed into a per-node temporary directory.
+    Twisted[tls,conch] >= 19.10.0
+
+    PyYAML >= 3.11
+
+    six >= 1.10.0
+
+    # for 'tahoe invite' and 'tahoe join'
+    magic-wormhole >= 0.10.2
+
+    # A new enough version to support custom JSON encoders.
+    eliot >= 1.13.0
+
+    # Pyrsistent 0.17.0 (which we use by way of Eliot) has dropped
+    # Python 2 entirely; stick to the version known to work for us.
+    pyrsistent
```

Comment:
Can we just drop this dependency then (and rely on Eliot declarations) since we don't support py2?

Reply:
Ah yes, probably so.